src/components/pages/collectRecordFormPages/HabitatComplexityForm/HabitatComplexityForm.js

```diff
@@ -114,8 +120,22 @@ const HabitatComplexityForm = ({ isNewRecord }) => {
     )
   }, [collectRecordBeingEdited, getPersistedUnsavedFormikData])
 
-  const handleCollectRecordChange = (updatedCollectRecord) =>
+  const handleCollectRecordChange = (updatedCollectRecord) => {
+    // maybe rename this function. Its pretty vague...
+    // need to recompute record name for sub nav node for all CRs
+    // setSubNavNode(recordNameForSubNode)
+
+    // { name: siteName, number: transectNumber, label }
+
+    setSubNavNode(
```

Comment:
haven't tested yet, but looking at the code, it seems like we are setting the sites data in the initial `useEffect` call and not accounting for future changes. so if any sites data (such as the name) is updated `onSave`, the habitat complexity component will still be relying on the initial state. We would have to fetch new sites data in any update, and specifically check if the site name was updated before doing so. lmk if that makes senses!

Adding the site name to the event info should be a simple fix in the api. we can still send the id - however it should be explicitly stated which is the id and which is the site name, instead of the generic `site` title.  while I understand there are tradeoffs we have to make for a data syncing app, I don't think this is one of them.

Reply:
agree that we can make the `getDataForSubNavNode` reusable for more components!